keanu-python/keanu/vertex/base.py

```diff
@@ -1,170 +1,78 @@
-
-    def __pow__(self, other):
-        return kn.vertex.generated.Power(self, other)
-
-    def __rpow__(self, other):
-        return kn.vertex.generated.Power(other, self)
-
-    def __truediv__(self, other):
-        return kn.vertex.generated.Division(self, other)
-
-    def __rtruediv__(self, other):
-        return kn.vertex.generated.Division(other, self)
 
-    def __floordiv__(self, other):
-        return kn.vertex.generated.IntegerDivision(self, other)
-
-    def __rfloordiv__(self, other):
-        return kn.vertex.generated.IntegerDivision(other, self)
-
-    def __eq__(self, other):
-        return kn.vertex.generated.Equals(self, other)
-
-    def __req__(self, other):
-        return kn.vertex.generated.Equals(self, other)
-
-    def __ne__(self, other):
-        return kn.vertex.generated.NotEquals(self, other)
-
-    def __rne__(self, other):
-        return kn.vertex.generated.NotEquals(self, other)
-
-    def __gt__(self, other):
-        return kn.vertex.generated.GreaterThan(self, other)
-
-    def __ge__(self, other):
-        return kn.vertex.generated.GreaterThanOrEqual(self, other)
-
-    def __lt__(self, other):
-        return kn.vertex.generated.LessThan(self, other)
-
-    def __le__(self, other):
-        return kn.vertex.generated.LessThanOrEqual(self, other)
-
-    def __abs__(self):
-        return kn.vertex.generated.Abs(self)
-
-    def __round__(self):
-        return kn.vertex.generated.Round(self)
-
-    def __floor__(self):
-        return kn.vertex.generated.Floor(self)
-
-    def __ceil__(self):
-        return kn.vertex.generated.Ceil(self)
+k = KeanuContext()
 
 
 class Vertex(JavaObjectWrapper, VertexOps):
-    def __init__(self, val, *args):
+    def __init__(self,
+                 val: Any,
+                 *args: Union[vertex_param_types, shape_types]) -> None:
         if args:
             ctor = val
             val = ctor(*(Vertex.__parse_args(args)))
 
         super(Vertex, self).__init__(val)
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(self.get_id())
 
-    def observe(self, v):
+    def observe(self, v: tensor_arg_types) -> None:
         self.unwrap().observe(Tensor(v).unwrap())
 
-    def set_value(self, v):
+    def set_value(self, v: tensor_arg_types) -> None:
         self.unwrap().setValue(Tensor(v).unwrap())
 
-    def set_and_cascade(self, v):
+    def set_and_cascade(self, v: tensor_arg_types) -> None:
         self.unwrap().setAndCascade(Tensor(v).unwrap())
 
-    def sample(self):
+    def sample(self) -> numpy_types:
         return Tensor._to_ndarray(self.unwrap().sample())
 
-    def get_value(self):
+    def get_value(self) -> numpy_types:
         return Tensor._to_ndarray(self.unwrap().getValue())
 
-    def get_connected_graph(self):
+    def get_connected_graph(self) -> Iterator['Vertex']:
         return Vertex._to_generator(self.unwrap().getConnectedGraph())
 
-    def get_id(self):
+    def get_id(self) -> Tuple[Any, ...]:
```

Comment:
Is the `id` a tuple?

Reply:
Yeah, id in java is an array. We make it a tuple here so it's immutable.